### PR TITLE
feat: UI implementation and application enhancements

### LINKED ISSUE #50: Enhancement: Complete UI
### Summary

UI should implement the functionality to call APIs

- Signup
- Login
- Profile
- Teams
- Team Members & Invitations

### Basic Example

_No response_

### Drawbacks and Impact

_No response_

### Unresolved questions

_No response_

### LINKED ISSUE #75: Bug: running "make lint" reformats the code
### Description

Running `make lint` seems to format the code.

After a `make lint`, the following files are modified:

```
	modified:   src/app/cli.py
	modified:   src/app/contrib/aiosql/service.py
	modified:   src/app/contrib/vite/config.py
	modified:   src/app/contrib/vite/loader.py
	modified:   src/app/domain/accounts/controllers/accounts.py
	modified:   src/app/domain/accounts/guards.py
	modified:   src/app/domain/accounts/services.py
	modified:   src/app/domain/plugins.py
	modified:   src/app/domain/tags/controllers.py
	modified:   src/app/domain/teams/guards.py
	deleted:    src/app/domain/web/public/.gitkeep
	modified:   src/app/lib/db/base.py
	modified:   src/app/lib/db/migrations/versions/2023-07-14_initial_models_ddc1b07aa56f.py
	modified:   src/app/lib/dependencies.py
	modified:   src/app/lib/dto.py
	modified:   src/app/lib/repository.py
	modified:   src/app/lib/service/sqlalchemy.py
	modified:   src/app/lib/settings.py
	modified:   src/app/lib/worker/controllers.py
	modified:   src/app/lib/worker/info.py
	modified:   src/app/lib/worker/plugin.py
	modified:   src/app/utils.py
	modified:   tests/conftest.py
	modified:   tests/docker_service.py
	modified:   tests/integration/conftest.py
	modified:   tests/integration/test_access.py
	modified:   tests/integration/test_accounts.py
	modified:   tests/integration/test_teams.py
	modified:   tests/unit/lib/test_crypt.py
	modified:   tests/unit/lib/test_db.py
	modified:   tests/unit/lib/test_dependencies.py
	modified:   tests/unit/test_cli.py
```

I think it's too surprising to have a "make lint" action reformat the code. 

### URL to code causing the issue

_No response_

### MCVE

_No response_

### Steps to reproduce

```bash
Type `make lint`
```

### Screenshots

_No response_

### Logs

_No response_

### Litestar Version

The one specified in the project

### Platform

- [ ] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

### LINKED ISSUE #105: Bug: schema typescript command failing
### Description

On clear pull of the repo and after running `make install`. Trying to run `app schema typescript` results in an error
The error displayed do not give me enough information to find the root issue

### URL to code causing the issue

_No response_

### MCVE

```python
# Your MCVE code here
```

### Steps to reproduce

```bash
1. Pull main branch
2. Install using `make install`
3. Activate venv
4. Run `app schema typescript` command
5. See error
```

### Screenshots

"In the format of: `SCREENSHOT_DESCRIPTION`"

### Logs

```bash
Traceback (most recent call last):
  File "/project/litestar-fullstack/.venv/bin/app", line 8, in <module>
    sys.exit(run_cli())
             ^^^^^^^^^
  File "/project/litestar-fullstack/src/app/__main__.py", line 23, in run_cli
    run_litestar_cli()
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/__main__.py", line 6, in run_cli
    litestar_group()
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/rich_click/rich_command.py", line 126, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/cli/_utils.py", line 280, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/cli/_utils.py", line 280, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/cli/commands/schema.py", line 81, in generate_typescript_specs
    specs = convert_openapi_to_typescript(app.openapi_schema, namespace)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/_openapi/typescript_converter/converter.py", line 291, in convert_openapi_to_typescript
    parse_request_body(
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/_openapi/typescript_converter/converter.py", line 205, in parse_request_body
    parse_schema(schema) if body.required else TypeScriptUnion((parse_schema(schema), undefined)),
    ^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/_openapi/typescript_converter/schema_parsing.py", line 152, in parse_schema
    return parse_type_schema(schema=schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/litestar-fullstack/.venv/lib/python3.12/site-packages/litestar/_openapi/typescript_converter/schema_parsing.py", line 129, in parse_type_schema
    raise TypeError(f"received an unexpected openapi type: {schema.type}")  # pragma: no cover
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: received an unexpected openapi type: None
```

### Package Version

Main branch of litestar-fullstack

### Platform

- [ ] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

### PR DESCRIPTION
**Pull request by cofin**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

Includes the following changes:
- Uses latest `litestar-vite` integration
- Refactors models into a single directory to prevent circular imports
- Adds a separate `config` module to separate `app/lib` classes from configuration
- Converts to React based UI
- Implements shadui components

### Close Issue(s)

 - Closes #75 
 - Closes #50 
 - Closes #105

### CODE REVIEW

**Comment by JacobCoffee** [comment]
PR is so big my browser keeps crashing.. ill try to review more later!

File: .vscode/settings.json [hunk 1]
```diff
@@ -19,67 +19,52 @@
...
+  "python.terminal.executeInFileDir": true,
+  "python.testing.pytestEnabled": true,
+  "autoDocstring.guessTypes": false,
+  "python.analysis.autoImportCompletions": true,
+  "python.analysis.autoFormatStrings": true,
+  "python.analysis.extraPaths": ["${workspaceFolder}/src/server"],
   "editor.formatOnSave": true,
-  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "notebook.formatOnSave.enabled": true,
+  "black-formatter.args": ["--line-length=120"],
```
[the remark this answers is not in the document]
Comment (cofin) [reply .vscode/settings.json:41]: nope, that can likely be removed.

**Comment by v3ss0n** [comment]
I am going to test it out this week.
Chose react instead of view?

**Comment by JacobCoffee** [comment]
i say lets get it passing in Railway deploys and merge. I literally cannot open it on my desktop or my mac to review, it just crashes my browsers

**Review by JacobCoffee — APPROVED** [review]

**Pull request merged**